RecoLocalCalo/HGCalRecAlgos/src/HGCalImagingAlgo.cc

```diff
@@ -4,60 +4,71 @@
 #include "Geometry/Records/interface/IdealGeometryRecord.h"
 #include "Geometry/CaloGeometry/interface/CaloSubdetectorGeometry.h"
 #include "Geometry/CaloGeometry/interface/CaloCellGeometry.h"
+#include "DataFormats/HcalDetId/interface/HcalSubdetector.h"
 
 #include "RecoEcal/EgammaCoreTools/interface/PositionCalc.h"
 //
 #include "DataFormats/CaloRecHit/interface/CaloID.h"
 
+void HGCalImagingAlgo::populate(const HGCRecHitCollection& hits){
+  //loop over all hits and create the Hexel structure, skip energies below ecut
 
-// Create a vector of Hexels associated to one cluster from a collection of HGCalRecHits - this can be used 
-// directly to make the final cluster list - this method can be invoked multiple times for the same event 
-// with different input (reset should be called between events)
-void HGCalImagingAlgo::makeClusters(const HGCRecHitCollection& hits)
-{
-  //used for speedy search 
-  std::vector<std::vector<KDNode> >points(2*(maxlayer+1));
-  std::vector<KDTree> hit_kdtree(2*(maxlayer+1));
-
-  std::vector<std::array<float,2> > minpos(2*(maxlayer+1),{ {0.0f,0.0f} }),maxpos(2*(maxlayer+1),{ {0.0f,0.0f} });
-
-  //  std::vector<std::vector<Hexel> > points(2*(maxlayer+1)); //a vector of vectors of hexels, one for each layer
-  //@@EM todo: the number of layers should be obtained programmatically - the range is 1-n instead of 0-n-1...
-
-
-  if (verbosity < pINFO)
-    {
-      std::cout << "-------------------------------------------------------------" << std::endl;
-      std::cout << "HGC Imaging algorithm invoked for " << std::endl;
-      std::cout << "delta_c " << delta_c << " kappa " << kappa;
-      //      if( doSharing ) std::cout << " showerSigma " << std::sqrt(sigma2);
-      std::cout << std::endl;
-    }
+  if (dependSensor) {
+  // for each layer and wafer calculate the thresholds (sigmaNoise and energy) once
+    computeThreshold();
+  }
 
-  //loop over all hits and create the Hexel structure, skip energies below ecut
   for (unsigned int i=0;i<hits.size();++i) {
+
     const HGCRecHit& hgrh = hits[i];
-    if(hgrh.energy() < ecut) continue; 
     DetId detid = hgrh.detid();
+    unsigned int layer = rhtools_.getLayerWithOffset(detid);
+    float thickness = 0.;
+    // set sigmaNoise default value 1 to use kappa value directly in case of sensor-independent thresholds
+    float sigmaNoise = 1.;
+    if(dependSensor){
+      if (layer<= lastLayerFH) // only EE and FH have silicon sensors
+        thickness = rhtools_.getSiThickness(detid);
+      double storedThreshold=thresholds[layer-1][layer<=lastLayerFH ? rhtools_.getWafer(detid) : 0];
```

Comment:
Hi @perrotta - since you would like the indentation to be fixed, which is a mixture or tabs and spaces (and varies between the individual files) since most likely previous developers of the code did not set up their text editors correctly, do you mind me running `clang-format --style=Google -i` or something similar on the files that are part of this PR (caveat being that a lot more lines will show up as changed)? Alternatively, I can replace all tabs by 2 spaces. I'm open for suggestions.